tck/features/WithAcceptance.feature

```diff
@@ -0,0 +1,361 @@
+      """
+      CREATE (a {name: 'A'}),
+        (b {name: 'B'}),
+        (c {name: 'C'})
+      """
+    When executing query:
+      """
+      MATCH (a)
+      WITH a
+      WHERE a.name = 'B'
+      RETURN a
+      """
+    Then the result should be:
+      | a             |
+      | ({name: 'B'}) |
+    And no side effects
+
+  Scenario: WHERE after WITH can filter on top of an aggregation
+    Given an empty graph
+    And having executed:
+      """
+      CREATE (a {name: 'A'}),
+        (b {name: 'B'})
+      CREATE (a)-[:REL]->()
+      CREATE (a)-[:REL]->()
+      CREATE (a)-[:REL]->()
+      CREATE (b)-[:REL]->()
+      """
+    When executing query:
+      """
+      MATCH (a)-->()
+      WITH a, count(*) AS relCount
+      WHERE relCount > 1
+      RETURN a
+      """
+    Then the result should be:
+      | a             |
+      | ({name: 'A'}) |
+    And no side effects
+
+  Scenario: ORDER BY on an aggregating key
+    Given an empty graph
+    And having executed:
+      """
+      CREATE (a1 {bar: 'A'}),
+        (a2 {bar: 'A'}),
+        (b {bar: 'B'})
+      """
+    When executing query:
+      """
+      MATCH (a)
+      WITH a.bar AS bars, count(*) AS relCount
+      ORDER BY a.bar
+      RETURN *
+      """
+    Then the result should be:
+      | bars | relCount |
+      | 'A'  | 2        |
+      | 'B'  | 1        |
+    And no side effects
+
+  Scenario: ORDER BY a DISTINCT column
+    Given an empty graph
+    And having executed:
+      """
+      CREATE (a1 {bar: 'A'}),
+        (a2 {bar: 'A'}),
+        (b {bar: 'B'})
+      """
+    When executing query:
+      """
+      MATCH (a)
+      WITH DISTINCT a.bar AS bars
+      ORDER BY a.bar
+      RETURN *
+      """
+    Then the result should be:
+      | bars |
+      | 'A'  |
+      | 'B'  |
+    And no side effects
+
+  Scenario: WHERE on a DISTINCT column
+    Given an empty graph
+    And having executed:
+      """
+      CREATE (a1 {bar: 'A'}),
+        (a2 {bar: 'A'}),
+        (b {bar: 'B'})
+      """
+    When executing query:
+      """
+      MATCH (a)
+      WITH DISTINCT a.bar AS bars
+      WHERE a.bar = 'B'
+      RETURN *
+      """
+    Then the result should be:
+      | bars |
+      | 'B'  |
+    And no side effects
+
+  Scenario: A simple pattern with one bound endpoint
+    Given an empty graph
+    And having executed:
+      """
+      CREATE (a:A)-[:REL]->(b:B)
+      """
+    When executing query:
+      """
+      MATCH (a:A)-[r:REL]->(b:B)
+      WITH a AS b, b AS tmp, r AS r
+      WITH b AS a, r
+      LIMIT 1
+      MATCH (a)-[r]->(b)
+      RETURN a, r, b
+      """
+    Then the result should be:
+      | a    | r      | b    |
+      | (:A) | [:REL] | (:B) |
+    And no side effects
+
+  Scenario: Null handling
+    Given an empty graph
+    When executing query:
+      """
+      OPTIONAL MATCH (a:Start)
+      WITH a
+      MATCH (a)-->(b)
+      RETURN *
+      """
+    Then the result should be:
+      | a | b | r |
```

Comment:
I agree this is better. I think we ought to distinguish between nothing being returned (such as  CREATE statement) versus an empty result set being returned. Will cardify so in existing TCK files...